### PR TITLE
ARROW-6683: [Python] Test for fastparquet <-> pyarrow cross-compatibility

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
https://issues.apache.org/jira/browse/ARROW-6683

@wesm is this more or less what you were thinking?

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
The fast parquet test should run in the Buildbot conda builders now.

**Comment by jorisvandenbossche** [comment]
> The fast parquet test should run in the Buildbot conda builders now.

Indeed:

```
...
pyarrow/tests/test_parquet.py::test_fastparquet_cross_compatibility PASSED [ 80%]
...
```

**Review by kszucs — APPROVED** [review]
LGTM

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -3345,3 +3345,41 @@ def test_filter_before_validate_schema(tempdir):
...
+
+    df = pd.DataFrame(
+        {
+            "a": list("abc"),
+            "b": list(range(1, 4)),
+            "c": np.arange(4.0, 7.0, dtype="float64"),
+            "d": [True, False, True],
+            "e": pd.date_range("20130101", periods=3),
+            "f": pd.Categorical(["a", "b", "a"])
+            # "g": [[1, 2], None, [1, 2, 3]],
```
Comment (pitrou) [inline python/pyarrow/tests/test_parquet.py]: I assume it's commented out because it currently doesn't work? Add a comment describing why?

Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_parquet.py]: Yes, indeed, I was first trying it, but apparently fastparquet writes such column as BYTE_ARRAY JSON (while we write is as a actual nested INT64 column). 
I added a comment

**Pull request closed without merging**